### ISSUE TITLE
'gt' / "go to tab" functionnality is confused after moving tabs around

### ISSUE THREAD

**Issue by sim590** [The-Compiler/qutebrowser#2141]
Use 'gm' to move a tab for instance and bring the list of buffers up with 'gt' and see the indexes for each tabs. Some of them will be shifted (depending where you moved a tab).

**Comment by The-Compiler** [comment]
When doing something else to update it (like opening a new tab), things look right again - so I think it just doesn't update itself when a tab is moved.

**Comment by Raihid** [comment]
I've looked at the code and I think you're right. The tab completion class has all these signals attached which trigger rebuilding.

`qutebrowser/completion/models/miscmodels.py, class TabCompletionModel`
```
for i in range(tabbed_browser.count()):
    tab = tabbed_browser.widget(i)
    tab.url_changed.connect(self.rebuild)
    tab.title_changed.connect(self.rebuild)
    tab.shutting_down.connect(self.delayed_rebuild)
tabbed_browser.new_tab.connect(self.on_new_tab)
```
So it should be as easy as adding another signal for tab movement. Thing is, tab movement is handled by `CommandDispatcher` class, while all the other signals are in domain of `TabbedBrowser`. So if you want to create a signal that emits when a tab is moved, you could add a signal to `CommandDispatcher` and emit it at the end of `tab_move` -- but then you have to make `CommandDispatcher` inherit from `QObject`, which seems kinda hacky.

**Comment by The-Compiler** [comment]
So I just noticed [QTabBar](http://doc.qt.io/qt-5/qtabbar.html) (which is accessible via `tabbed_browser.tabBar()` has a `moveTab` method and a `tabMoved` signal.

The best solution would probably be to change `:tab-move` to use `moveTab`, and trigger a rebuild on the `tabMoved` signal.

**Comment by Raihid** [comment]
It should be fixed by #2152.
Not sure why this wasn't closed automatically, since the commit message contains both "Fixed" and "#2141". Probably I shouldn't have put those parentheses around the issue number.

**Comment by The-Compiler** [comment]
You'll need to mention "Fixed #2141" somewhere (in the body of a commit or the PR for example)
